staging/src/k8s.io/apiserver/pkg/endpoints/request/requestinfo.go

```diff
@@ -206,8 +206,8 @@ func (r *RequestInfoFactory) NewRequestInfo(req *http.Request) (*RequestInfo, er
 		requestInfo.Resource = requestInfo.Parts[0]
 	}
 
-	// if there's no name on the request and we thought it was a get before, then the actual verb is a list or a watch
-	if len(requestInfo.Name) == 0 && requestInfo.Verb == "get" {
+	// if there's either no name nor no namespace on the request and we thought it was a get before, then the actual verb is a list or a watch
+	if (len(requestInfo.Name) == 0 || len(requestInfo.Namespace) == 0) && requestInfo.Verb == "get" {
```

Comment:
This doesn't seem correct.

The pattern "/api/v1/nodes/foo" would hit that condition, but it isn't list/watch, but a regular get.
In other words, cluster-scoped resources with name set.